cylc/flow/parsec/fileparse.py

```diff
@@ -343,6 +347,43 @@ def merge_template_vars(
         return native_tvars
 
 
+def _prepend_old_templatevars(fpath: str, template_vars: t.Dict) -> t.Dict:
+    """If the fpath is in a rundir, extract template variables from database.
+
+    Args:
+        fpath: filepath of workflow config file.
+        template_vars: Template vars to prepend old tvars to.
+    """
+    rundir = Path(fpath).parent
+    old_tvars = OldTemplateVars(rundir).template_vars
+    old_tvars.update(template_vars)
+    template_vars = old_tvars
+    return template_vars
```

Comment:
Put this in [cylc/flow/templatevars.py](https://github.com/cylc/cylc-flow/pull/5214/files#diff-489533e95911c3f09c836f60cfb0a618358460ddd1b626b91e154d5b6789b986)?